app.py

```diff
@@ -186,25 +185,29 @@ class MQStatus(object):
             logger.error("handle_file(): unable to decode msg as json: {}".format(msg.value))
             continue
 
-        hash_ = data['hash']
+        if data.get('payload_id'):
+            payload_id = data['payload_id']
+        else:
+            payload_id = data.get('hash')
         result = data['validation']
 
-        logger.info('processing message: %s - %s', hash_, result)
+        logger.info('processing message: %s - %s', payload_id, result)
         if result.lower() == 'success':
             url = await IOLoop.current().run_in_executor(
-                None, storage.copy, storage.QUARANTINE, storage.PERM, hash_
+                None, storage.copy, storage.QUARANTINE, storage.PERM, payload_id
```

Comment:
What if the payload doesn’t contain neither _payload_id_, nor _hash_? It’s still data from an external service. Do we just want pass _None_ to _storage.copy_? Maybe it’d be good to raise an exception or log an error instead. Currently it takes down the whole service.